apps/store/extensions/app/social-reviews/themes/store/js/custom/review.js

```diff
@@ -247,11 +258,99 @@ $stream.on('click', '#btn-delete', function(e) {
+                    var icon = $("<i/>", {
+                        class: "fw fw-info"
+                    });
+                    $firstReview.find('h4').empty().append(icon).append(msg);
+                    msg = "Tell others what you think about this " + assetType;
+                    $firstReview.find('p').html(msg);
+                }
+            });
         }
     });
 });
-$reviews_frame.niceScroll();
+$container.on('click', '#btn-update', function (e) {
+    showLoading(true);
+    e.preventDefault();
+    var rating = $('.user-rating > .selected-rating').data().rating;
+    var review = getTextArea().val().trim();
+    var id = $(this).data('id');
+    var url = caramel.url('/apis/user-review/' + id);
+    var activityData = {
+        verb: 'review',
+        isMyComment: true,
+        object: {
+            objectType: 'review',
+            id: id,
+            rating: rating,
+            content: review
+        },
+        target: {
+            id: target
+        }
+    };
+    $.ajax({
+        url: url,
+        type: 'PUT',
+        contentType: 'application/json',
+        data: JSON.stringify(activityData),
+        success: function (updatedActivity) {
+            myReview = updatedActivity;
+            usingTemplate(function (template) {
+                var newComment = template(updatedActivity);
+                getMyReviewElement().html('').html(newComment);
+            });
+            showLoading(false);
+        }
+    });
+});
+$container.on('click', '#btn-edit', function (e) {
+    showLoading(true);
+    var myComment = getMyReviewElement();
+    var rating = myComment.find('.com-rating').data('rating');
+    var review = myComment.find('.com-content').html().trim();
+    var id = myComment.find('.com-review').data('target-id');
+    var editable = {
+        rating: rating,
+        review: review,
+        id: id
+    };
+    e.preventDefault();
+    $('.com-review[data-target-id="' + id + '"]').fadeOut();
+    caramel.partials({
+        comment: '/extensions/app/social-reviews/themes/' + caramel.themer + '/partials/comment-input.hbs'
+    }, function () {
+        var template = Handlebars.partials['comment'];
+        var commentInput = template(
+            {myReview: null, type: target.split(':')[0], ratings: [5, 4, 3, 2, 1], editable: editable});
+        $(".row.com-add").replaceWith(commentInput);
+    });
+});
+$container.on('click', '#btn-cancel', function (e) {
+    showLoading(true);
+    usingTemplate(function (template) {
+        var newComment = template(myReview);
+        getMyReviewElement().html('').html(newComment);
+    });
+    showLoading(false);
+});
+$container.on('click', '.rating-star', function () {
+    getAlertElement().html('').fadeOut("fast");
+    $('#btn-post.disabled').removeClass('disabled');
+    $(this).siblings().removeClass("selected-rating");
+    $(this).addClass("selected-rating");
+});
+$stream.niceScroll();
```

Comment:
AFAIU review.js is a very important script, therefore, I believe documentation plays a vital role when its comes to maintainability. Kindly document at least important functions.